### PR TITLE
update lingui config and add pre-commit

### PR DESCRIPTION
**Pull request by vipineth**
- Update .linguirc to not tract line number in `.po` files to remove unnecessary conflicts
- Add all messages.js (compiled file) to .gitignore
- Add lingui extract  to `pre-commit` hook

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202948224553666

### CODE REVIEW

**Review by divhead — DISMISSED** [review]

**Comment by gkrasulya** [comment]
thanks for good description 🙂 it helps

**Comment by gdev8317** [comment]
thanks for good description :) it helps

File: package.json [hunk 1]
```diff
@@ -100,7 +101,7 @@
     "prettier": "2.5.1"
   },
   "lint-staged": {
-    "**/*.js?(x)": [
+    "**/*.{js, jsx, ts, tsx}": [
```
Comment (gdev8317) [inline package.json:104]: does it work okay with spaces? afaik native sh/bash won't allow it. probably it's different syntax here, just wondering

**Review by gdev8317 — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (vipineth) [reply package.json:104]: It makes sense to up remove the spaces. I will update to `src/**/*.{js,ts,jsx,tsx}`

**Pull request merged**